code/game/gamemodes/umbrage/umbrage.dm

```diff
@@ -0,0 +1,184 @@
+/*
+
+Umbrage is the revamped Shadowling.
+The gamemode focuses around several powerful aliens, called umbrages, that attempt to enslave crew members.
+The umbrages thrive in darkness and can see in it, but the light physically hurts them.
+They start disguised with no special abilities, and have to spend time "hatching", during which they are vulnerable.
+
+Every umbrage on the station shares a goal: "ascend" into a progenitor, their true form.
+Resembling eldritch monstrosities, progenitors are immensely powerful, immune to most damage and capable of killing with a mere thought.
+When an umbrage successfully ascends, the shuttle is called. This is their win condition.
+In order to do this, they must accumulate a certain amount of veils based on round population.
+Ascension itself takes a long time, and the umbrage is completely vulnerable. Their veils must protect them.
+To obtain this, umbrages and their veils are linked through silent hive mind. This can be detected with the right machinery.
+
+Fluff-wise, umbrages are much more deceptive than shadowlings, and less brute-force. They think themselves more refined than humans rather than simply superior.
+Think of how we look on apes, for instance. Similar to us, and what we came from, but stupid and slow compared to us.
+They believe themselves as using magic, but it's more mundane; their bodies are capable of the things they believe are magic.
+
+Terminology you might remember has been changed to spookier variants, such as:
+	shadowlings => umbrages (pronounced "um-bridge")
+	thralls => veils
+	ascendants => progenitors
+
+Folder contents:
+	/code/game/gamemodes/umbrage
+		/__umbrage_defines.dm: Contains defines and global variables for all umbrage code.
+		/umbrage.dm: This file. Contains the skeleton of the gamemode.
+		/umbrage_antags.dm: Contains antagonist datums for umbrages and veils.
+		/umbrage_datum.dm: Contains the tracking datum for lots of umbrage stuff.
+		/umbrage_major_abilities.dm: Contains hatching and ascension abilities and their related objects.
+		/umbrage_minor_abilities.dm: Contains spells, abilities, and other stuff. Does NOT including hatching and ascension.
+		/umbrage_objects.dm: Contains the few items and structures used in Umbrage.
+		/umbrage_unsorted.dm: Contains things with nowhere else to be, like the umbrage race.
+
+Idea and initial code by Xhuis (my 3rd gamemode now...)
+
+*/
+
+///////////////
+// FRAMEWORK //
+///////////////
+
+/datum/game_mode
+	var/list/umbrages = list() //A list of the minds of all umbrages, including progenitors.
+	var/list/veils = list() //A list of the minds of all veils.
+	var/list/umbrages_and_veils = list() //A list of the minds of all umbrages and veils.
```

Comment:
Because the Mindlink checks who's an umbrage or veil every time someone speaks, I figured it'd be faster to just keep track in general and not fetch from both lists. Or should I just do that?